### PR TITLE
Issue #1346: Prevent php notices when comments are hidden.

### PR DESCRIPTION
**Pull request by jenlampton**
Fixes https://github.com/backdrop/backdrop-issues/issues/1346

### CODE REVIEW

File: core/modules/comment/comment.module [hunk 1]
```diff
@@ -724,7 +724,7 @@ function comment_node_page_additions(Node $node) {
   // Only attempt to render comments if the node has visible comments.
   // Unpublished comments are not included in $node->comment_count, so show
   // comments unconditionally if the user is an administrator.
-  if (($node->comment_count && user_access('access comments')) || user_access('administer comments')) {
+  if ((property_exists($node, 'comment_count') && $node->comment_count && user_access('access comments')) || user_access('administer comments')) {
```
Comment (quicksketch) [inline core/modules/comment/comment.module]: Instead of

```
property_exists($node, 'comment_count') && $node->comment_count
```

We could shorten this to:

```
!empty($node->comment_count)
```

Would that be better?

Comment (jenlampton) [inline core/modules/comment/comment.module]: yes. Fixed.

File: core/modules/comment/comment.module [hunk 2]
```diff
@@ -724,7 +724,7 @@ function comment_node_page_additions(Node $node) {
   // Only attempt to render comments if the node has visible comments.
   // Unpublished comments are not included in $node->comment_count, so show
   // comments unconditionally if the user is an administrator.
-  if (($node->comment_count && user_access('access comments')) || user_access('administer comments')) {
+  if ((!empty($node->comment_count) && $node->comment_count && user_access('access comments')) || user_access('administer comments')) {
```
Comment (quicksketch) [inline core/modules/comment/comment.module]:
```
!empty($node->comment_count) && $node->comment_count
```

The second conditional here is redundant with the first one. It can be simplified down to just:

```
!empty($node->comment_count)
```

**Comment by jenlampton** [comment]
Fixed, rebased and updated.

**Comment by quicksketch** [comment]
Merged into 1.x and 1.2.x. Thanks!
